### PR TITLE
Fix JSON output when exporting custom emojis

### PR DESCRIPTION
**Pull request by EtaoinWu**
Attempts to fix issue #24961.

### CODE REVIEW

**Comment by EtaoinWu** [comment]
Setting up a build env is too much of a hassle for me, so this is untested. If the github actions were approved I can probably test that

File: Telegram/SourceFiles/export/output/export_output_json.cpp [hunk 1]
```diff
@@ -194,10 +194,10 @@ QByteArray SerializeText(
 			: (part.type == Type::TextUrl)
 			? "href"
 			: "none";
-		const auto additionalValue = (part.type == Type::MentionName
-			|| part.type == Type::CustomEmoji)
+		const auto additionalValue = (part.type == Type::MentionName)
 			? part.additional
-			: (part.type == Type::Pre || part.type == Type::TextUrl)
+			: (part.type == Type::Pre || part.type == Type::TextUrl
+			|| part.type == Type::CustomEmoji)
```
Comment (23rd) [inline Telegram/SourceFiles/export/output/export_output_json.cpp:200]:
```suggestion
			: (part.type == Type::Pre || part.type == Type::TextUrl
				|| part.type == Type::CustomEmoji)
```

File: Telegram/SourceFiles/export/output/export_output_json.cpp [hunk 2]
```diff
@@ -194,10 +194,10 @@ QByteArray SerializeText(
 			: (part.type == Type::TextUrl)
 			? "href"
 			: "none";
-		const auto additionalValue = (part.type == Type::MentionName
-			|| part.type == Type::CustomEmoji)
+		const auto additionalValue = (part.type == Type::MentionName)
 			? part.additional
-			: (part.type == Type::Pre || part.type == Type::TextUrl)
+			: (part.type == Type::Pre || part.type == Type::TextUrl
```
Comment (john-preston) [inline Telegram/SourceFiles/export/output/export_output_json.cpp:199]:
```suggestion
			: (part.type == Type::Pre
				|| part.type == Type::TextUrl
```

**Review by john-preston — COMMENTED** [review]
I've approved the actions so I'd be glad if you can test it.

**Comment by EtaoinWu** [comment]
I have built c4090b7 locally and tested it, and it resolved the issue. (All the later commits are just reformats.)

**Pull request merged**